project_closing/models/project.py

```diff
@@ -9,8 +9,7 @@ class ProjectProject(models.Model):
     def toggle_active(self):
         if not self.env.context.get('doing_project_toggle_active'):
             # When called directly from Project, delegate to Analytic Account
-            res = self.analytic_account_id.toggle_active()
+            return self.analytic_account_id.toggle_active()
         else:
             # When called from the Analytic Account, perform the toggling
-            res = super(ProjectProject, self).toggle_active()
-        return res
+            return super(ProjectProject, self).toggle_active()
```

Comment:
Sure its one LOC less, but I prefer to have a single exit point as much as possible. But then, its a subjective preference.

Reply:
True, and I mostly agree with that. Here's an excerpt from "Clean Code" which I like very much:

"Some programmers follow Edsger Dijkstra’s rules of structured programming. Dijkstra said that every function, and every block within a function, should have one entry and one exit. [...] those rules serve little benefit when functions are very small. [...] So if you keep your functions small, then the occasional multiple return, break, or continue statement does no harm and can sometimes even be more expressive than the single-entry, single-exit rule."

Subjective, true. I think this "exception" holds true mainly for strongly typed, nestable/narrow variable scope languages (C#, Java) where those principles make such a method a bit too verbose. Still, I found those 4 lines very small, and more expressive :)

In all cases, let's never do that in a 20+ lines method...